cache/refresher.go

```diff
@@ -66,15 +73,48 @@ func handleSegmentMessage(_ context.Context, msg domain.SSEMessage) error {
 	return nil
 }
 
-func handleProxyMessage(_ context.Context, msg domain.SSEMessage) error {
+func (s Refresher) handleProxyMessage(ctx context.Context, msg domain.SSEMessage) error {
 	switch msg.Event {
 	case domain.EventProxyKeyDeleted:
+		// todo
+		return nil
 	case domain.EventEnvironmentAdded:
+		if err := s.handleAddEnvironmentEvent(ctx, msg.Environments); err != nil {
+			s.log.Error(err.Error())
+			return err
+		}
 	case domain.EventEnvironmentRemoved:
+		// todo
+		return nil
 	case domain.EventAPIKeyAdded:
+		// todo
+		return nil
 	case domain.EventAPIKeyRemoved:
+		// todo
+		return nil
 	default:
 		return fmt.Errorf("%w %q for Proxymessage", ErrUnexpectedEventType, msg.Event)
 	}
 	return nil
 }
+
+// handleAddEnvironmentEvent fetches proxyConfig for all added environments and sets them on.
+func (s Refresher) handleAddEnvironmentEvent(ctx context.Context, environments []string) error {
+	for _, env := range environments {
+		input := domain.GetProxyConfigInput{
+			Key:               s.proxyKey,
+			EnvID:             env,
+			AuthToken:         s.authToken,
+			ClusterIdentifier: s.clusterIdentifier,
+			PageNumber:        0,
+			PageSize:          10,
+		}
+		// fetch the proxy config.
+		_, err := s.clientService.PageProxyConfig(ctx, input)
+		if err != nil {
+			s.log.Error("unable to fetch config for the environment: ", env)
```

Comment:
I don't think this will log out properly because this logger expects a message and key values. You should be able to change it to this though and it should work

```go
s.log.Error("unable to fetch config for the environment", "environment", env)
```